src/store/useTemplatesStore.ts

```diff
@@ -0,0 +1,197 @@
+			const templateMap: Record<string, TemplateMetadataWithCategory> = {};
+
+			await Promise.all(
+				processedCategories.map(async ({ cards, name }) => {
+					const processedCards = await Promise.all(
+						cards.map((card) => processTemplateCard(card, name, templateStorage))
+					);
+					processedCards.forEach((cardData) => {
+						templateMap[cardData.assetDirectory] = cardData;
+					});
+				})
+			);
+
+			const categories = Object.values(templateMap).reduce((acc, template) => {
+				const category = acc.find((c) => c.name === template.category);
+				if (category) {
+					category.templates.push(template);
+				} else {
+					acc.push({ name: template.category, templates: [template] });
+				}
+
+				return acc;
+			}, [] as TemplateCategory[]);
+
+			return { templateMap, categories };
+		};
+
+		try {
+			let shouldFetchTemplates = false;
+			let shouldFetchTemplatesFromGithub = false;
+			let lastCommitDate = get().lastCommitDate;
+			const currentTime = new Date();
+			const lastCheckDate = get().lastCheckDate;
+
+			const shouldCheckGitHub =
+				!lastCheckDate ||
+				currentTime.getTime() - new Date(lastCheckDate).getTime() >= templatesUpdateCheckInterval;
+
+			if (shouldCheckGitHub) {
+				try {
+					const { data } = await axios.get<GitHubCommit[]>(remoteTemplatesRepositoryURL, {
+						params: { per_page: 1 },
+						headers: { Accept: "application/vnd.github.v3+json" },
+					});
+
+					const latestCommit = data?.[0];
+
+					if (latestCommit) {
+						const latestCommitDate = latestCommit.commit.author.date;
+						const currentCommitDate = get().lastCommitDate;
+
+						if (currentCommitDate && new Date(latestCommitDate) <= new Date(currentCommitDate)) {
+							return;
+						}
+						shouldFetchTemplatesFromGithub = true;
+						shouldFetchTemplates = true;
+						lastCommitDate = latestCommitDate;
+					}
+					set({ lastCheckDate: currentTime });
+				} catch {
+					shouldFetchTemplates = true;
+				}
+			}
+
+			if (!shouldFetchTemplates) {
+				set({ isLoading: false });
+
+				return;
+			}
+
+			const zipUrl = shouldFetchTemplatesFromGithub ? remoteTemplatesArchiveURL : localTemplatesArchiveFallback;
+
+			const { categories, templateMap } = await processTemplates(zipUrl);
+			const sortedCategories = sortCategories(categories, templateCategoriesOrder);
+
+			set({
+				templateMap,
+				sortedCategories,
+				lastCommitDate,
+				isLoading: false,
+				error: null,
+			});
+
+			return;
+		} catch (error) {
+			const uiErrorMessage = i18n.t("templates.failedToFetch", { ns: "stores" });
+			let logErrorMessage = uiErrorMessage;
+			if (axios.isAxiosError(error)) {
+				logErrorMessage = i18n.t("templates.failedToFetchExtended", {
+					ns: "stores",
+					error: error?.response?.data,
+				});
+			} else {
+				logErrorMessage = i18n.t("templates.failedToFetchExtended", {
+					ns: "stores",
+					error: error?.message,
+				});
+			}
+			console.error(logErrorMessage);
```

Comment:
Why is not Logger?

Reply:
Basically we need only console.
But in logger I added parameter `consoleOnly` and passed `true` now.